Build failed when CONFIG_STM32_ARM_MPU_ENABLE=y

Following merge of #9776, building STM32 platform is failed when STM32_ARM_MPU_ENABLE is enabled.

Build error reported:
```
zephyr/arch/arm/core/cortex_m/mpu/libarch__arm__core__cortex_m__mpu.a(arm_mpu.c.obj): In function `arm_mpu_init':
/local/mcu/zephyr/zephyr-project/arch/arm/core/cortex_m/mpu/arm_mpu.c:368: undefined reference to `mpu_config'
```

`mpu_config` is defined in `soc/arm/st_stm32/common/arm_mpu_regions.c` which is not compiled since soc/arm/st_stm32/common/CMakeLists.txt is not parsed anymore

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -336,7 +336,7 @@ add_subdirectory(misc)
 # property which is set implicitly for custom command outputs
 include(misc/generated/CMakeLists.txt)
 
-if(EXISTS soc/${ARCH}/CMakeLists.txt)
+if(EXISTS ${SOC_DIR}/${ARCH}/CMakeLists.txt)
   add_subdirectory(${SOC_DIR}/${ARCH} soc/${ARCH})
 else()
   add_subdirectory(${SOC_DIR}/${ARCH}/${SOC_PATH} soc/${ARCH}/${SOC_PATH})

